quickwit/quickwit-metastore/src/metastore/postgresql_model.rs

```diff
@@ -210,3 +211,84 @@ impl TryInto<DeleteTask> for PgDeleteTask {
         })
     }
 }
+
+#[derive(sqlx::Type, PartialEq, Debug)]
+#[sqlx(type_name = "SHARD_STATE")]
+#[sqlx(rename_all = "lowercase")]
+pub enum PgShardState {
+    Unspecified,
+    Open,
+    Unavailable,
+    Closed,
+}
+
+impl From<PgShardState> for String {
```

Comment:
I think we can get rid of this and use [`as_json_str_name`](https://github.com/quickwit-oss/quickwit/blob/main/quickwit/quickwit-proto/src/ingest/mod.rs#L195) directly when binding values to queries.